src/thread.cr

```diff
@@ -50,7 +50,9 @@ class Thread
   # All threads, so the GC can see them (GC doesn't scan thread locals)
   @@threads = Set(Thread).new
 
+  {% if !flag?(:openbsd) %}
   @[ThreadLocal]
```

Comment:
Maybe not memoizing `@@current` on OpenBSD, then iterating `@@threads` to find the current thread that matches the id returned by `pthread_self`?